core/src/main/java/org/apache/accumulo/core/client/impl/TableOperationsImpl.java

```diff
@@ -1209,6 +1209,15 @@ public void online(String tableName) throws AccumuloSecurityException, AccumuloE
   public void online(String tableName, boolean wait) throws AccumuloSecurityException, AccumuloException, TableNotFoundException {
     checkArgument(tableName != null, "tableName is null");
     String tableId = Tables.getTableId(context.getInstance(), tableName);
+
+    /**
+     * ACCUMULO-4574 if table is already online return without executing fate operation.
+     */
+    TableState expectedState = Tables.getTableState(context.getInstance(), tableId);
```

Comment:
The implementation of this uses zoo cache.   So it may return stale information.  Before getting the table state, could call `Tables.clearCache()`.  This will force the table state to be read from zookeeper.